src/js/trove/table.js

```diff
@@ -248,6 +364,114 @@
           }, "multi-order");
         }),
 
+        'add-column': runtime.makeMethod2(function(_, colname, eltList) {
+          ffi.checkArity(3, arguments, "add-column", true);
+          runtime.checkString(colname);
+          runtime.checkList(eltList);
+          var asArray = runtime.ffi.toArray(eltList);
+          if(rows.length !== asArray.length) {
+            throw runtime.ffi.throwMessageException("column-length-mismatch");
+          }
+
+          var newRows = [];
+          for(var i = 0; i < rows.length; i += 1) {
+            newRows.push(rows[i].concat([asArray[i]]));
+          }
+          
+          return makeTable(headers.concat([colname]), newRows);
+        }),
+
+        'add-row': runtime.makeMethod1(function(_, row) {
+          ffi.checkArity(2, arguments, 'add-row', true);
+          runtime.checkRow(row);
+          var theseKeys = Object.keys(headerIndex);
+          var rowKeys = Object.keys(row.$underlyingTable.headerIndex);
+          if(theseKeys.length !== rowKeys.length) {
+            throw runtime.ffi.throwMessageException("add-row-length");
+          }
+          for(var i = 0; i < theseKeys.length; i += 1) {
+            if(headerIndex["column:" + headers[i]] !== 
+                row.$underlyingTable.headerIndex["column:" + headers[i]]) {
+              throw runtime.ffi.throwMessageException("row-name-mismatch: " + headers[i]);
```

Comment:
Names must match up in the correct positions, so if you append an `a,b,c` row to a `a,c,b` table, it's an error.